### PR TITLE
CloudWatch: Fix strict Typescript errors

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Fixes some --strict errors in the CloudWatch plugin.

One remains, related to rxjs which I'm not familiar enough to fix...

**Which issue(s) this PR fixes**:

Relates to #32142  
Relates to #40460

**Special notes for your reviewer**:

Can you believe it?!?? Just TWO errors remaining!!!!!!!

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx [hunk 1]
```diff
@@ -71,10 +71,16 @@ function useAuthenticationWarning(jsonData: CloudWatchJsonData) {
...
 function useDatasource(datasourceName: string) {
   const [datasource, setDatasource] = useState<CloudWatchDatasource>();
+
   useEffect(() => {
     getDatasourceSrv()
       .loadDatasource(datasourceName)
-      .then((datasource: CloudWatchDatasource) => setDatasource(datasource));
+      .then((datasource) => {
+        // It's really difficult to type .loadDatasource() because it's inherently untyped as it involves two JSON.parse()'s
+        // So a "as" type assertion here is a nessisary evil.
```
Comment (sarahzinger) [inline public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx:80]:
```suggestion
        // So a "as" type assertion here is a necessary evil.
```

**Review by sarahzinger — APPROVED** [review]
neat!!!

File: public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx (hunk 1 above)
Comment (joshhunt) [reply public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx]: awkward

**Comment by joshhunt** [comment]
The one remaining error for CloudWatch is in `cloudwatch/utils/rxjs/increasingInterval.ts`:

https://github.com/grafana/grafana/blob/cce7341afab20c0afe1482aba318b77cb16f3bdf/public/app/plugins/datasource/cloudwatch/utils/rxjs/increasingInterval.ts#L13


Our signature for `dispatch` is  

```ts
(this: SchedulerAction<IntervalState>, state: IntervalState) => void
```

but typescript / `scheduler.schedule` says that the second `state` argument can potentially be undefined.

Now, at the moment if `state` is undefined, it'll result in a runtime exception (because it tries to unpack values from an undefined object), but I'm not sure the right way to fix this because I'm unfamiliar with rxjs and observables.

**Pull request merged**